code/main/core/identityModule.brs

```diff
@@ -53,16 +51,15 @@ function _adb_IdentityModule(configurationModule as object) as object
         end function,
 
         updateECID: function(ecid as dynamic) as void
-            if ecid <> invalid and ecid = m._ecid
-                _adb_logVerbose("updateECID() - Not updating ECID. Same value is cached and persisted.")
-                return
-            end if
+            m._ecid = ecid
             if ecid = invalid
                 _adb_logDebug("updateECID() - Deleting ECID.")
+                m._deleteECID()
+            else
+                _adb_logVerbose("updateECID() - Saving ECID:(" + FormatJson(ecid) + ") in cache and persistence.")
+                m._saveECID(m._ecid)
             end if
-            _adb_logVerbose("updateECID() - Saving ECID:(" + FormatJson(ecid) + ") in cache and persistence.")
-            m._ecid = ecid
-            m._saveECID(m._ecid)
+
         end function,
 
         _loadECID: function() as dynamic
```

Comment:
Not related to your change, can we check if the loaded ECID is not empty before returning?

Reply:
I have refactored it and that would be the responsibility of the caller to check and log.